libs/utils/platforms/hikey960_energy.py

```diff
@@ -0,0 +1,119 @@
+a53_cluster_idle_states = OrderedDict([
+    ("WFI",               12),
+    ("cpu-sleep-0",       12),
+    ("cluster-sleep-0",   0),
+])
+
+a53_cpu_active_states = OrderedDict([
+    (533000, ActiveState(capacity=133, power=87)),
+    (999000, ActiveState(capacity=250, power=164)),
+    (1402000, ActiveState(capacity=351, power=265)),
+    (1709000, ActiveState(capacity=429, power=388)),
+    (1844000, ActiveState(capacity=462, power=502)),
+])
+
+a53_cpu_idle_states = OrderedDict([
+    ("WFI",               5),
+    ("cpu-sleep-0",       0),
+    ("cluster-sleep-0",   0),
+])
+
+a53s = [0, 1, 2, 3]
+
+def a53_cpu_node(cpu):
+    return EnergyModelNode(cpu=cpu,
+                           active_states=a53_cpu_active_states,
+                           idle_states=a53_cpu_idle_states)
+
+a73_cluster_active_states = OrderedDict([
+    (903000, ActiveState(power=102)),
+    (1421000, ActiveState(power=124)),
+    (1805000, ActiveState(power=221)),
+    (2112000, ActiveState(power=330)),
+    (2362000, ActiveState(power=433)),
+])
+
+a73_cluster_idle_states = OrderedDict([
+    ("WFI",               102),
+    ("cpu-sleep-0",       102),
+    ("cluster-sleep-0",   0),
+])
+
+a73_cpu_active_states = OrderedDict([
+    (903000, ActiveState(capacity=390, power=404)),
+    (1421000, ActiveState(capacity=615, power=861)),
+    (1805000, ActiveState(capacity=782, power=1398)),
+    (2112000, ActiveState(capacity=915, power=2200)),
+    (2362000, ActiveState(capacity=1024, power=2848)),
+])
+
+a73_cpu_idle_states = OrderedDict([
+    ("WFI",               18),
+    ("cpu-sleep-0",       0),
+    ("cluster-sleep-0",   0),
+])
+
+a73s = [4, 5, 6, 7]
+
+def a73_cpu_node(cpu):
+    return EnergyModelNode(cpu=cpu,
+                           active_states=a73_cpu_active_states,
+                           idle_states=a73_cpu_idle_states)
+
+hikey960_energy = EnergyModel(
+    root_node=EnergyModelRoot(
+        children=[
+            EnergyModelNode(
+                name="cluster_a53",
+                active_states=a53_cluster_active_states,
+                idle_states=a53_cluster_idle_states,
+                children=[a53_cpu_node(c) for c in a53s]),
+            EnergyModelNode(
+                name="cluster_a73",
+                active_states=a73_cluster_active_states,
+                idle_states=a73_cluster_idle_states,
+                children=[a73_cpu_node(c) for c in a73s])]),
+    root_power_domain=PowerDomain(idle_states=[], children=[
+        PowerDomain(
+            idle_states=["cluster-sleep-0"],
+            children=[PowerDomain(idle_states=["WFI", "cpu-sleep-0"], cpu=c)
+                      for c in a73s]),
+        PowerDomain(
+            idle_states=["cluster-sleep-0"],
+            children=[PowerDomain(idle_states=["WFI", "cpu-sleep-0"], cpu=c)
+                      for c in a53s])]),
+    freq_domains=[a53s, a73s])
+
+class SanityCheck(_StaticEnergyModelTest):
```

Comment:
That looks a bit awkward, I suppose that's here because you want to run a test by executing something like `lisa-test libs/utils/platforms/hikey960_energy.py` and you can't give arguments to nosetest other than the test file...

What about having a test in lisa/tests/lisa that just runs the SanityCheck against all of our platform energy files? On the plus side, that'll also make Travis run that against PRs.

Reply:
The issue with having Travis look at that is that all platforms have at least one test failing ... For good reasons. And Elieva already implemented some really nice stuff in `test/lisa` I think

Other than that, I quite like the idea of being able to do `lisa-test libs/utils/platforms/hikey960_energy.py:SanityCheck` for example. I don't think the goal of these tests is to actually get them to pass. It is just to report information giving us an idea of how broken a platform is. So I liked the idea to be able to run the test on individual platforms. I assume that most of the time we'll run these tests only once, when we have a new platform

Reply:
Okay then. The only bit I don't like is having to redefine the test method for each model, but that's because of how we define those EMs. And that's also why we end up doing all those lines of `import *_energy` in TestEnv.

Anyway, that's nothing this PR is supposed to address, so I guess that's fine.

Reply:
We agreed offline that the fact that we don't really want these tests to pass is simply because they're not really tests ... So I'll remove the `_StaticEnergyModelTest` abstract class and add a notebook instead